### PR TITLE
M10

### PR DESCRIPTION
**Pull request by valeria1604**

### CODE REVIEW

File: src/com/company/M10/Task2.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        List<User> users = readFromFile(inputFileName);
+        writeToFile(outputFileName, users);
+    }
+
+    private static List<User> readFromFile(String inputFileName) {
+        List<User> users = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(inputFileName))) {
+            String line = reader.readLine();
+            while ((line = reader.readLine()) != null){
+                String[] data = line.split(" ");
```
Comment (maxline) [inline src/com/company/M10/Task2.java]: create a constant for hardcoded strings

File: src/com/company/M10/Task2.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                if(data.length == 2){
+                    users.add(new User(data[0], Integer.parseInt(data[1])));
```
Comment (maxline) [inline src/com/company/M10/Task2.java]: create a constants for hardcoded values 0 ,1, 2

File: src/com/company/M10/Task2.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return users;
+    }
+
+    private static void writeToFile(String outputFileName, List<User> users) {
+//        Gson gson = new Gson();
```
Comment (maxline) [inline src/com/company/M10/Task2.java]: implement writing json to the file
use these examples:
https://stackoverflow.com/questions/29319434/how-to-save-data-with-gson-in-a-json-file
https://www.baeldung.com/gson-save-file

File: src/com/company/M10/Task3.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//        }
+//
+//        System.out.println(words);
+    }
+
+    private static void calculateRepetitions(Map<String, Integer> words, String line) {
+        String[] data = line.split("\\s+");
+        System.out.println(Arrays.toString(data));
+        for (String word : data) {
+            if (!words.containsKey(word)) {
```
Comment (maxline) [inline src/com/company/M10/Task3.java]: why this `if` section needed? is not  code in `else` section enough?

File: src/com/company/M10/Task3.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+//                calculateRepetitions(words, line);
+//            }
+//        } catch (IOException ex) {
+//            ex.printStackTrace();
+//        }
+//
+//        System.out.println(words);
+    }
+
+    private static void calculateRepetitions(Map<String, Integer> words, String line) {
```
Comment (maxline) [inline src/com/company/M10/Task3.java]: It is better if method will return a result of calculation. Map<String, Integer> will be good choice.
Think do we really need a map as method parameter?

File: src/com/company/M10/Task3.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package com.company.M10;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class Task3 {
+    public static void main(String[] args) {
+//        String sentence = "the day is sunny the the\n" +
```
Comment (maxline) [inline src/com/company/M10/Task3.java]: why this code is commented?

File: src/com/company/M10/Task3.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+//                "the sunny is is";
+//        Map<String, Integer> words = new HashMap<>();
+//        calculateRepetitions(words, sentence);
+
+//        String inputFileName = "file.txt";
+//        Map<String, Integer> words = new HashMap<>();
+//        try (BufferedReader reader = new BufferedReader(new FileReader(inputFileName))) {
+//            String line;
+//            while ((line = reader.readLine()) != null) {
```
Comment (maxline) [inline src/com/company/M10/Task3.java]: according the task we should calculate count words in whole file, not needed calculate its in every line.

**Review by maxline — CHANGES_REQUESTED** [review]
Good job! Lets improve the solution

File: src/com/company/M10/Task2.java (hunk 1 above)
Comment (valeria1604) [reply src/com/company/M10/Task2.java]: fixed

File: src/com/company/M10/Task2.java (hunk 2 above)
Comment (valeria1604) [reply src/com/company/M10/Task2.java]: fixed

File: src/com/company/M10/Task2.java (hunk 3 above)
Comment (valeria1604) [reply src/com/company/M10/Task2.java]: fixed

File: src/com/company/M10/Task3.java (hunk 5 above)
Comment (valeria1604) [reply src/com/company/M10/Task3.java]: fixed

File: src/com/company/M10/Task3.java (hunk 6 above)
Comment (valeria1604) [reply src/com/company/M10/Task3.java]: fixed

File: src/com/company/M10/Task3.java (hunk 7 above)
Comment (valeria1604) [reply src/com/company/M10/Task3.java]: fixed

File: src/com/company/M10/Task2.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Task2 {
+
```
Comment (maxline) [inline src/com/company/M10/Task2.java]:
```suggestion
```

File: src/com/company/M10/Task2.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    public static final String SPACE = " ";
+    public static final int NAME_PLACE = 0;
+    public static final int AGE_PLACE = 1;
+    public static final int TWO_VALUES = 2;
```
Comment (maxline) [inline src/com/company/M10/Task2.java]: let's give move accurate names for the constants that explain business logic, e.g.:
```suggestion
    public static final String DELIMITER = " ";
    public static final int NAME_INDEX = 0;
    public static final int AGE_INDEX = 1;
    public static final int COLUMNS_COUNT = 2;
```

File: src/com/company/M10/Task2.java (hunk 3 above)
Comment (maxline) [reply src/com/company/M10/Task2.java]: why this code is commented?
it is not good practice push commented code.

File: src/com/company/M10/Task3.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        Map<String, Integer> words = calculateWordFrequencies(inputFileName);
+        System.out.println(words);
+    }
+
+    private static Map<String, Integer> calculateWordFrequencies(String inputFileName) {
+        Map<String, Integer> words = new HashMap<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(inputFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] data = line.split("\\s+");
```
Comment (maxline) [inline src/com/company/M10/Task3.java]: create a constant for the hardcoded string. think about good naming

**Review by maxline — CHANGES_REQUESTED** [review]
good job! left a couple more suggestions. take a look please

File: src/com/company/M10/Task2.java (hunk 8 above)
Comment (valeria1604) [reply src/com/company/M10/Task2.java]: fixed

File: src/com/company/M10/Task3.java (hunk 10 above)
Comment (valeria1604) [reply src/com/company/M10/Task3.java]: fixed

**Review by maxline — APPROVED** [review]
Well done! Approved

**Pull request merged**